mllib/src/main/scala/org/apache/spark/ml/clustering/LDA.scala

```diff
@@ -574,18 +578,28 @@ object LocalLDAModel extends MLReadable[LocalLDAModel] {
       val metadata = DefaultParamsReader.loadMetadata(path, sc, className)
       val dataPath = new Path(path, "data").toString
       val data = sparkSession.read.parquet(dataPath)
-        .select("vocabSize", "topicsMatrix", "docConcentration", "topicConcentration",
-          "gammaShape")
-        .head()
-      val vocabSize = data.getAs[Int](0)
-      val topicsMatrix = data.getAs[Matrix](1)
-      val docConcentration = data.getAs[Vector](2)
-      val topicConcentration = data.getAs[Double](3)
-      val gammaShape = data.getAs[Double](4)
+      val vectorConverted = MLUtils.convertVectorColumnsToML(data, "docConcentration")
+      val Row(vocabSize: Int, topicsMatrix: Matrix, docConcentration: Vector,
+      topicConcentration: Double, gammaShape: Double) = MLUtils.convertMatrixColumnsToML(
```

Comment:
style: indent 4 spaces

You could also simplify this with Datasets by using ```(DataFrame).as[Data]``` if you make ```LocalLDAModelWriter.Data``` accessible here (by moving it to sit under ```object LocalLDAModel```).